src/python/zquantum/core/hamiltonian.py

```diff
@@ -103,6 +109,18 @@ def get_decomposition_function(
     return decomposition_function
 
 
+def _calculate_variance_upper_bound(group: QubitOperator) -> float:
+    coefficients = np.array(list(group.terms.values()))
+    return np.sum(coefficients ** 2)
+
+
+def _remove_constant_term_from_group(group: QubitOperator) -> QubitOperator:
+    new_group = copy.deepcopy(group)
+    if new_group.terms.get(()):
+        del new_group.terms[()]
+    return new_group
```

Comment:
What happens if the only term in the group is the constant term, as seems to be the case for `group_comeasurable_terms_greedy` above?

Reply:
It works (added a test for that in https://github.com/zapatacomputing/z-quantum-core/pull/268/commits/17e16d8ed08d4f29bb0f4e872990b354f685d1e0)